### DISCUSSION TITLE
AES/CBC 256/pcks7 using ESP 32 s3 IDF

### DISCUSSION THREAD

**Discussion by RuchaVs** [community/community#76758] [Programming Help]
### Select Topic Area

Question

### Body

Hello,

I am trying to implement encryption/Decryption in ESP 32 S3 using their IDF.The Sample code i referred to is 
[https://github.com/Mbed-TLS/mbedtls/blob/development/programs/psa/crypto_examples.c#L300](url), which works fine . 
 Basically my problem statement is ,i am getting a encrypted string in Base 64 format, which i need to decrypt, do some processing, encrypt it and send it back.
Currently , i am facing a problem that if a string other than ascii  format is passed to encryption /decryption function its throws a error (-135) , i.e Invalid parameter.

i wanted some help to cope up with the above problem.

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @RuchaVs ! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the GitHub product Discussions.

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.
